specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+                        << L"Reason:\t"              << ProcessFailedReasonToString(reason)    << L"\n"
+                        << L"Exit code:\t"           << std::to_wstring(exitCode)              << L"\n"
+                        << L"Process description:\t" << processDescription.get()               << std::endl;
+                MessageBox(
+                    m_appWindow->GetMainWindow(), message.str().c_str(),
+                    L"Child process failed", MB_OK);
+            }
+            return S_OK;
+        })
+        .Get(),
+    &m_processFailedToken));
+//! [ProcessFailed]
+
+```
+
+## .NET C#
+```xml
+<Window x:Class="WebView2WpfBrowser.MainWindow"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+        xmlns:wv2="clr-namespace:Microsoft.Web.WebView2.Wpf;assembly=Microsoft.Web.WebView2.Wpf"
+        xmlns:local="clr-namespace:WebView2WpfBrowser"
+        x:Name="MyWindow"
+        Title="MainWindow"
+        Height="450"
+        Width="800"
+>
+
+    <!-- ... -->
+
+    <DockPanel
+        x:Name="MyDockPanel"
+    >
+        <!-- ... -->
+
+        <DockPanel DockPanel.Dock="Top">
+
+            <!-- ... -->
+
+            <TextBox x:Name="url" Text="{Binding ElementName=webView,Path=Source,Mode=OneWay}">
+                <TextBox.InputBindings>
+                    <KeyBinding Key="Return" Command="NavigationCommands.GoToPage" CommandParameter="{Binding ElementName=url,Path=Text}" />
+                </TextBox.InputBindings>
+            </TextBox>
+        </DockPanel>
+
+        <wv2:WebView2
+            x:Name="webView"
+            CreationProperties="{StaticResource EvergreenWebView2CreationProperties}"
+            Source="https://www.bing.com/"
+            NavigationStarting="WebView_NavigationStarting"
+            NavigationCompleted="WebView_NavigationCompleted"
+        />
+    </DockPanel>
+</Window>
+
+```
+
+```c#
+// This re-instantiates the control and attaches properties as set in the XAML
+// element. Replace once the control has reinit/uninitialize logic.
+void ReinitializeWebView()
+{
+    webView = new WebView2();
+
+    // Restore URI and other WebView state/setup.
+    webView.CreationProperties = (CoreWebView2CreationProperties)this.FindResource("EvergreenWebView2CreationProperties");
+    webView.NavigationStarting += WebView_NavigationStarting;
+    webView.NavigationCompleted += WebView_NavigationCompleted;
+
+    Binding urlBinding = new Binding()
+    {
+        Source = webView,
+        Path = new PropertyPath("Source"),
+        Mode = BindingMode.OneWay
+    };
+    url.SetBinding(TextBox.TextProperty, urlBinding);
+
+    MyDockPanel.Children.Add(webView);
+    webView.Source = _uriToRestore ?? new Uri("https://www.bing.com");
+}
+
+async void WebView_ProcessFailed(CoreWebView2 sender, CoreWebView2ProcessFailedEventArgs e)
```

Comment:
Should show this method being registered for the ProcessFailed event.

Reply:
Adding in `WebView_CoreWebView2InitializationCompleted` definition.